colossalai/inference/core/engine.py

```diff
@@ -3,8 +3,10 @@
 from typing import Dict, List, Optional, Tuple, Type, Union
 
 import numpy as np
+import PIL.Image
```

Comment:
**About the design of Engine**: 

I recommend to split the two engine instances into two files, saying that, LLMEngine and DiffusionEngine, which inherit from a base abstract class. In this way, you are able to:
1. retain some abstraction level and split the logics of executing into different places
2. instantiate certain engine during initialization
3. **prevent dispatching step method** (and other methods applying conditions) every time of inference

Reply:
This will apply to modifications along this file

Reply:
thanks!